nixos/modules/services/web-apps/netbox.nix

```diff
@@ -139,6 +142,8 @@ in {
   };
 
   config = mkIf cfg.enable {
+    services.netbox.plugins = mkIf cfg.enableLdap (ps: [ ps.django-auth-ldap ]);
```

Comment:
It does not seem it will merge well, alas :/.
Enabling LDAP will prevent anyone from adding any plugin to this list.
I'd advise to write an explicit `merge` function for the option, maybe?

Reply:
From my tests it does seem to merge well:

```nix
{config, lib, pkgs, ...}: {
  options.thing = lib.mkOption {
    type = with lib.types; functionTo (listOf package);
    default = ps: [];
  };

  options.thingPkgs = lib.mkOption {
    type = with lib.types; listOf package;
  };

  config.thing = lib.mkMerge [
    (ps: [ps.hello])
    (ps: [ps.htop])
  ];

  config.thingPkgs = config.thing pkgs;
}
```

`config.thingPkgs` does evaluate to `[ <<hello>> <<htop>> ]`

From my sight reading of the source of the `functionTo` type, I think it create a function that applies the argument to all definitions, and merges the the result using the element type merge function, which should be what we want.